### PR TITLE
Add support for the `bulk.write.ordered` sink connector property

### PR DESCRIPTION
**Pull request by stIncMale**
KAFKA-253

### CODE REVIEW

File: src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java [hunk 1]
```diff
@@ -353,37 +446,63 @@ void configureCapturing(
...
+                        cast(ArgumentCaptor.forClass(BulkWriteOptions.class));
+                    verify(collection, atLeast(0))
+                        .bulkWrite(modelsCaptor.capture(), optionsCaptor.capture());
+                    List<List<? extends WriteModel<? extends BsonDocument>>> models =
+                        modelsCaptor.getAllValues();
+                    List<BulkWriteOptions> options = optionsCaptor.getAllValues();
+                    assertEquals(models.size(), options.size());
+                    int bulkWritesCount = options.size();
+                    return IntStream.range(0, bulkWritesCount)
+                        .mapToObj(i -> new CapturedBulkWrite(models.get(i), options.get(i)))
```
Comment (stIncMale) [inline src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java:460]: Poor man's zip operation.

File: src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java [hunk 2]
```diff
@@ -189,35 +212,90 @@ void putTolerateAllOrderedWriteError() {
             asList(
                 // batch1
                 Records.simpleValid(TEST_TOPIC, 0),
+                Records.simpleValid(TEST_TOPIC, 1),
```
Comment (stIncMale) [inline src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java:215]: Changed the first batch, which is successful, such that it contains multiple records. As a result, indices in all other batches were shifted.

File: src/main/java/com/mongodb/kafka/connect/sink/dlq/AnalyzedBatchFailedWithBulkWriteException.java [hunk 3]
```diff
@@ -108,7 +110,6 @@ public void log() {
     if (!skippedRecords.isEmpty()) {
       logger.log(skippedRecords.values(), writeSkippedException);
     }
-    logger.log(null, e);
```
Comment (stIncMale) [inline src/main/java/com/mongodb/kafka/connect/sink/dlq/AnalyzedBatchFailedWithBulkWriteException.java:111]: This message was moved to `StartedMongoSinkTask.handleTolerableWriteException` for the following reason: the log message `"Failed to put into the sink the following records: null ... exception details"` is weird and confusing. Specifying a meaningful message from here would require having another method on the `AnalyzedBatchFailedWithBulkWriteException.Logger` interface.

`StartedMongoSinkTask.handleTolerableWriteException` now logs the `MongoBulkWriteException`, without analyzing it and specifying which records actually failed to be written. Then `AnalyzedBatchFailedWithBulkWriteException.log` logs each failed record with a more specific exception relevant to it.

**Review by rozza — APPROVED** [review]
LGTM

**Pull request merged**